src/core/QueryManager.ts

```diff
@@ -1398,11 +1416,23 @@ export class QueryManager {
       // WatchQueryOptions object.
       normalized.variables = variables;
 
+      const cacheWriteBehavior =
+        fetchPolicy === "no-cache" ? CacheWriteBehavior.FORBID
+          // Watched queries must opt into overwriting existing data on refetch,
+          // by passing refetchWritePolicy: "overwrite" in their WatchQueryOptions.
+        : (
+          networkStatus === NetworkStatus.refetch &&
+          normalized.refetchWritePolicy !== "merge"
+        ) ?
+          CacheWriteBehavior.OVERWRITE
+        : CacheWriteBehavior.MERGE;
```

Comment:
```suggestion
          normalized.refetchWritePolicy === "merge"
        ) ?
          CacheWriteBehavior.MERGE
        : CacheWriteBehavior.OVERWRITE;
```

Since this is checking against `merge` directly, we should make that part of the "if" instead of the "else". I know this was written this way before, but since you're touching the code, might as well make this a touch more readable.

Reply:
This code already existed before and I just moved it from `fetchQueryByPolicy` to  `fromVariables` - I have to admit that no matter how we turn this, it doesn't completely seem to match up with the comment 

> Watched queries must opt into overwriting existing data on refetch, by passing refetchWritePolicy: "overwrite" in their WatchQueryOptions.

since it seems to actually default to overriding and needs an opt-in into `merge`.

Truth said, I'd prefer to not touch this any more during this PR. This probably needs a good look and an update of the comment - and then we get to that, I'd be more open to flipping conditions around.